docker-compose/oidc/README.adoc

```diff
@@ -0,0 +1,361 @@
+= Enable Unified Identity with Azure Entra ID for Redpanda and Redpanda Console
+:env-docker: true
+:page-categories: Security, Management, Redpanda Console
+:description: Integrate Azure Entra ID with Redpanda and Redpanda Console for unified identity using OpenID Connect (OIDC).
+:page-layout: lab
+// Set up attributes to hold the latest version of Redpanda and Redpanda Console.
+// For GitHub, hard-code the latest version to these values:
+ifndef::env-site[]
+:latest-redpanda-version: 25.1.1
+:latest-console-version: 3.1.0
+endif::[]
+// For the docs site, use the built-in attributes that store the latest version as fetched from GitHub releases.
+ifdef::env-site[]
+:latest-redpanda-version: {full-version}
+// All pages already have access to {latest-console-version} on the docs site.
+endif::[]
+
+This lab walks you through integrating Azure Entra ID (formerly Azure AD) with Redpanda and Redpanda Console for unified identity using OpenID Connect (OIDC). This integration allows you to use Azure Entra ID for authentication and authorization in Redpanda and Redpanda Console.
+
+== Prerequisites
+
+- Access to https://portal.azure.com[Azure Portal^]
+- https://learn.microsoft.com/en-us/entra/identity-platform/quickstart-create-new-tenant[Azure Entra ID tenant^]
+ifndef::env-site[]
+- link:https://docs.redpanda.com/current/get-started/rpk-install/[Install `rpk`] on your host machine.
+endif::[]
+ifdef::env-site[]
+- xref:get-started:rpk-install.adoc[Install `rpk`] on your host machine.
+endif::[]
+- https://docs.docker.com/compose/install/[Docker and Docker Compose^] installed on your host machine.
+
+== Set up Azure Entra ID
+
+This section guides you through creating an Azure Entra ID app registration and configuring it for use with Redpanda.
+You'll set up the necessary permissions, scopes, and claims to ensure proper authentication and authorization.
+
+=== Create an app registration
+
+In this section, you'll register a new application in Entra ID that Redpanda will use to authenticate.
+
+. Sign in to https://portal.azure.com[Azure Portal^].
+. Navigate to *Azure Active Directory* > *App registrations* > *New registration*.
+. Configure:
+** *Name*: `RedpandaUnifiedIdentity`
+** *Supported account types*: Choose *Accounts in this organizational directory only*
+** *Redirect URI*: `http://localhost:8080/auth/callbacks/oidc`
+. Click *Register*.
+
+See also: https://learn.microsoft.com/en-us/azure/active-directory/develop/quickstart-register-app[Register an application^].
+
+=== Set access token version to v2
+
+Redpanda and Redpanda Console require OAuth 2.0-compliant JWT tokens for user authentication. When using OIDC, your IdP must issue JWTs. Redpanda Console uses these tokens to authenticate to Redpanda APIs through SASL/OAUTHBEARER or Bearer headers.
```

Comment:
OIDC-compliant JWT tokens. OAuth2's tokens are opaque, they can be JWT but an OAuth2 provider will not do any of the OIDC validations against the tokens.

Reply:
In fact, using OAuth2 with JWT tokens can be insecure, as Console does not have a way to validate the cryptographic signature of the token if it treats it as JWT. It may allow easy attacks through token manipulation.